doxygen/man/man3/dlgNewLink.3
.TH "dlgNewLink" 3 "cppMD01" \" -*- nroff -*-
.ad l
.nh
.SH NAME
dlgNewLink
.SH SYNOPSIS
.br
.PP
.PP
\fR#include <dlgNewLink\&.h>\fP
.PP
Inherits wxDialog\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBdlgNewLink\fP (wxWindow *parent, wxWindowID id=wxID_ANY, const wxPoint &pos=wxDefaultPosition, const wxSize &size=wxDefaultSize)"
.br
.ti -1c
.RI "virtual \fB~dlgNewLink\fP ()"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "wxButton * \fBbtnCancel\fP"
.br
.ti -1c
.RI "wxButton * \fBbtnInset\fP"
.br
.ti -1c
.RI "wxStaticText * \fBlblDet\fP"
.br
.ti -1c
.RI "wxStaticText * \fBlblLink\fP"
.br
.ti -1c
.RI "wxStaticText * \fBlblidlk1\fP"
.br
.ti -1c
.RI "wxStaticText * \fBlnlLinkName\fP"
.br
.ti -1c
.RI "wxTextCtrl * \fBedtDetDes\fP"
.br
.ti -1c
.RI "wxTextCtrl * \fBedtLink\fP"
.br
.ti -1c
.RI "wxTextCtrl * \fBedtLinkName\fP"
.br
.ti -1c
.RI "wxTextCtrl * \fBedtPrim\fP"
.br
.in -1c
.SS "Static Protected Attributes"

.in +1c
.ti -1c
.RI "static const long \fBID_STATICTEXT1\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_TEXTCTRL1\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_STATICTEXT2\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_TEXTCTRL2\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_STATICTEXT3\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_TEXTCTRL3\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_STATICTEXT4\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_TEXTCTRL4\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_BUTTON1\fP = wxNewId()"
.br
.ti -1c
.RI "static const long \fBID_BUTTON2\fP = wxNewId()"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line \fB12\fP of file \fBdlgNewLink\&.h\fP\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "dlgNewLink::dlgNewLink (wxWindow * parent, wxWindowID id = \fRwxID_ANY\fP, const wxPoint & pos = \fRwxDefaultPosition\fP, const wxSize & size = \fRwxDefaultSize\fP)"
zrdb\&.initField(mkVal,dlg\&.edtPrim);
.PP
Definition at line \fB31\fP of file \fBdlgNewLink\&.cpp\fP\&.
.SS "dlgNewLink::~dlgNewLink ()\fR [virtual]\fP"

.PP
Definition at line \fB84\fP of file \fBdlgNewLink\&.cpp\fP\&.
.SH "Member Data Documentation"
.PP 
.SS "wxButton* dlgNewLink::btnCancel"

.PP
Definition at line \fB20\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxButton* dlgNewLink::btnInset"

.PP
Definition at line \fB21\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxTextCtrl* dlgNewLink::edtDetDes"

.PP
Definition at line \fB26\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxTextCtrl* dlgNewLink::edtLink"

.PP
Definition at line \fB27\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxTextCtrl* dlgNewLink::edtLinkName"

.PP
Definition at line \fB28\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxTextCtrl* dlgNewLink::edtPrim"

.PP
Definition at line \fB29\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_BUTTON1 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB43\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_BUTTON2 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB44\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_STATICTEXT1 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB35\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_STATICTEXT2 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB37\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_STATICTEXT3 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB39\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_STATICTEXT4 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB41\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_TEXTCTRL1 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB36\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_TEXTCTRL2 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB38\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_TEXTCTRL3 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB40\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "const long dlgNewLink::ID_TEXTCTRL4 = wxNewId()\fR [static]\fP, \fR [protected]\fP"

.PP
Definition at line \fB42\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxStaticText* dlgNewLink::lblDet"

.PP
Definition at line \fB22\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxStaticText* dlgNewLink::lblidlk1"

.PP
Definition at line \fB24\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxStaticText* dlgNewLink::lblLink"

.PP
Definition at line \fB23\fP of file \fBdlgNewLink\&.h\fP\&.
.SS "wxStaticText* dlgNewLink::lnlLinkName"

.PP
Definition at line \fB25\fP of file \fBdlgNewLink\&.h\fP\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for cppMD01 from the source code\&.


dlgNewLink.cpp
#include "dlgNewLink.h"
#include "zDB01.h"
#include "curlStuff.h"

zDB01 db02;
curlStuff cs01;

//(*InternalHeaders(dlgNewLink)
#include <wx/intl.h>
#include <wx/string.h>
//*)

//(*IdInit(dlgNewLink)
const long dlgNewLink::ID_STATICTEXT1 = wxNewId();
const long dlgNewLink::ID_TEXTCTRL1 = wxNewId();
const long dlgNewLink::ID_STATICTEXT2 = wxNewId();
const long dlgNewLink::ID_TEXTCTRL2 = wxNewId();
const long dlgNewLink::ID_STATICTEXT3 = wxNewId();
const long dlgNewLink::ID_TEXTCTRL3 = wxNewId();
const long dlgNewLink::ID_STATICTEXT4 = wxNewId();
const long dlgNewLink::ID_TEXTCTRL4 = wxNewId();
const long dlgNewLink::ID_BUTTON1 = wxNewId();
const long dlgNewLink::ID_BUTTON2 = wxNewId();
//*)

BEGIN_EVENT_TABLE(dlgNewLink,wxDialog)
    //(*EventTable(dlgNewLink)
    //*)
END_EVENT_TABLE()

dlgNewLink::dlgNewLink(wxWindow* parent,wxWindowID id,const wxPoint& pos,const wxSize& size)
{
    //(*Initialize(dlgNewLink)
    wxBoxSizer* BoxSizer1;
    wxBoxSizer* BoxSizer2;
    wxBoxSizer* BoxSizer3;
    wxBoxSizer* BoxSizer4;

    Create(parent, id, _("Insert New Link"), wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE|wxSYSTEM_MENU|wxRESIZE_BORDER, _T("id"));
    SetClientSize(wxDefaultSize);
    Move(wxDefaultPosition);
    BoxSizer1 = new wxBoxSizer(wxHORIZONTAL);
    BoxSizer2 = new wxBoxSizer(wxVERTICAL);
    BoxSizer3 = new wxBoxSizer(wxVERTICAL);
    lblidlk1 = new wxStaticText(this, ID_STATICTEXT1, _("Master Key"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT1"));
    BoxSizer3->Add(lblidlk1, 0, wxALL|wxEXPAND, 1);
    edtPrim = new wxTextCtrl(this, ID_TEXTCTRL1, wxEmptyString, wxDefaultPosition, wxSize(331,24), wxTE_READONLY, wxDefaultValidator, _T("ID_TEXTCTRL1"));
    BoxSizer3->Add(edtPrim, 0, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
    lnlLinkName = new wxStaticText(this, ID_STATICTEXT2, _("Link Name"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT2"));
    BoxSizer3->Add(lnlLinkName, 0, wxALL|wxEXPAND, 1);
    edtLinkName = new wxTextCtrl(this, ID_TEXTCTRL2, _("Text"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_TEXTCTRL2"));
    BoxSizer3->Add(edtLinkName, 0, wxALL|wxEXPAND, 1);
    lblLink = new wxStaticText(this, ID_STATICTEXT3, _("Link"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT3"));
    BoxSizer3->Add(lblLink, 0, wxALL|wxEXPAND, 1);
    edtLink = new wxTextCtrl(this, ID_TEXTCTRL3, _("Text"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_TEXTCTRL3"));
    BoxSizer3->Add(edtLink, 0, wxALL|wxEXPAND, 1);
    lblDet = new wxStaticText(this, ID_STATICTEXT4, _("Detailed Description"), wxDefaultPosition, wxDefaultSize, 0, _T("ID_STATICTEXT4"));
    BoxSizer3->Add(lblDet, 0, wxALL|wxEXPAND, 1);
    edtDetDes = new wxTextCtrl(this, ID_TEXTCTRL4, _("Note."), wxDefaultPosition, wxSize(339,112), wxTE_MULTILINE, wxDefaultValidator, _T("ID_TEXTCTRL4"));
    BoxSizer3->Add(edtDetDes, 0, wxALL|wxEXPAND, 1);
    BoxSizer4 = new wxBoxSizer(wxHORIZONTAL);
    btnInset = new wxButton(this, ID_BUTTON1, _("Insert"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON1"));
    BoxSizer4->Add(btnInset, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
    btnCancel = new wxButton(this, ID_BUTTON2, _("Cancel"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON2"));
    BoxSizer4->Add(btnCancel, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
    BoxSizer3->Add(BoxSizer4, 0, wxALL|wxEXPAND, 1);
    BoxSizer2->Add(BoxSizer3, 1, wxALL|wxEXPAND, 5);
    BoxSizer1->Add(BoxSizer2, 1, wxALL|wxEXPAND, 5);
    SetSizer(BoxSizer1);
    BoxSizer1->Fit(this);
    BoxSizer1->SetSizeHints(this);

    Connect(ID_BUTTON1,wxEVT_COMMAND_BUTTON_CLICKED,(wxObjectEventFunction)&dlgNewLink::OnbtnInsetClick);
    Connect(ID_BUTTON2,wxEVT_COMMAND_BUTTON_CLICKED,(wxObjectEventFunction)&dlgNewLink::OnbtnCancelClick);
    //*)

///   zrdb.initField(mkVal,dlg.edtPrim);
    db02.initField(db02.strIdlk1,edtPrim);
    //if db02.
    db02.initField(cs01.clipUrl(), edtLinkName);
    db02.initField(cs01.cbc, edtLink);
}

dlgNewLink::~dlgNewLink()
{
    //(*Destroy(dlgNewLink)
    //*)
}


void dlgNewLink::OnbtnInsetClick(wxCommandEvent& event)
{
    db02.insertLink(db02.fieldToString(edtPrim), db02.fieldToString(edtLinkName), db02.fieldToString(edtLink), db02.fieldToString(edtDetDes));
    EndModal(wxID_OK);
}

void dlgNewLink::OnbtnCancelClick(wxCommandEvent& event)
{
    EndModal(wxID_CANCEL);
}
